internal/auth/doc.go

```diff
@@ -0,0 +1,17 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package auth provides common types and functions for aiding in authentication
+// within Juju. Currently this package provides our password logic for hashing
+// and encapsulating plain text passwords.
+//
+// When a component in Juju receives a plain text password from a user it should
+// be immediately wrapped in a Password type with NewPassword("mypassword").
```

Comment:
`The source password should be immediately zero'd. Where possible, passwords should never become a string, and either be a byte slice or rune slice to ensure that the memory can be zero'd.`

Reply:
We would need api changes to support this. Punting for the moment and we will address this another day.